specs/Print.md

```diff
@@ -0,0 +1,816 @@
+{
+  CoreWebView2PrintSettings printSettings = null;
+  printSettings = WebViewEnvironment.CreatePrintSettings();
+  printSettings.ShouldPrintBackgrounds = true;
+  printSettings.ShouldPrintHeaderAndFooter = true;
+
+  return printSettings;
+
+  // or
+  // Get PrintQueue for the selected printer and use GetPrintCapabilities() of PrintQueue from System.Printing
+  // to get the capabilities of the selected printer.
+  // Display the printer capabilities to the user along with the page settings.
+  // Return the user selected settings.
+}
+```
+
+## PrintToPdfStream
+
+You can use `PrintToPdfStream` method to display as a preview in a custom print dialog.
+The app can also use this API to configure the custom print preview dialog unlike confining
+print preview dialog to the webview2 control.
+
+```cpp
+// This example prints the Pdf data of the current web page to a stream.
+
+// Function to display current web page pdf data in a custom print preview dialog.
+static void DisplayPdfDataInPrintDialog(IStream* pdfData)
+{
+  // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+
+bool AppWindow::PrintToPdfStream()
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+
+  wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+  CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+
+  wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+  CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+  wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+  CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+
+  CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(
+      webView2_15->PrintToPdfStream(
+          printSettings2.get(),
+          Callback<ICoreWebView2PrintToPdfStreamCompletedHandler>(
+              [title = title.get(), this](HRESULT errorCode, IStream* pdfData) -> HRESULT
+              {
+                DisplayPdfDataInPrintDialog(pdfData);
+                std::wstring message = L"Printing " + std::wstring(title) +
+                                       L" document to PDF Stream " +
+                                       ((errorCode == S_OK) ? L"succedded" : L"failed");
+
+                AsyncMessageBox(message, L"Print to PDF Stream");
+
+                return S_OK;
+              })
+              .Get()));
+  return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+  CoreWebView2PrintSettings printSettings = null;
+  printSettings = WebViewEnvironment.CreatePrintSettings();
+  printSettings.ShouldPrintBackgrounds = true;
+
+  string title = webView.CoreWebView2.DocumentTitle;
+
+  System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
```

Comment:
Inconsistent error checking between C++ and C#. Who is right?

Reply:
Please see below comment at the end about error handling.